aws/src/main/java/org/apache/iceberg/aws/s3/S3OutputStream.java

```diff
@@ -91,9 +104,36 @@ public void close() throws IOException {
     try {
       stream.close();
 
-      s3.putObject(
-          PutObjectRequest.builder().bucket(location.bucket()).key(location.key()).build(),
-          RequestBody.fromFile(stagingFile));
+      PutObjectRequest.Builder requestBuilder = PutObjectRequest.builder()
```

Comment:
Q: Why isn't the parent `super.close()` inside the try block as well?

Reply:
It's called on line 101

Reply:
Ah sorry misread the question. Technically `super.close()` calls `PositionOutputStream.close()` calls `OutputStream.close()` which does nothing, so I think it is fine to not have it in the try block.